lib/compat/wordpress-6.1/class-wp-theme-json-6-1.php

```diff
@@ -87,4 +87,132 @@ protected static function get_blocks_metadata() {
 
 		return static::$blocks_metadata;
 	}
+
+	/**
+	 * Builds metadata for the style nodes, which returns in the form of:
+	 *
+	 *     [
+	 *       [
+	 *         'path'     => [ 'path', 'to', 'some', 'node' ],
+	 *         'selector' => 'CSS selector for some node',
+	 *         'duotone'  => 'CSS selector for duotone for some node'
+	 *       ],
+	 *       [
+	 *         'path'     => ['path', 'to', 'other', 'node' ],
+	 *         'selector' => 'CSS selector for other node',
+	 *         'duotone'  => null
+	 *       ],
+	 *     ]
+	 *
+	 * @since 5.8.0
+	 *
+	 * @param array $theme_json     The tree to extract style nodes from.
+	 * @param array $selectors      List of selectors per block.
+	 * @return array
+	 */
+	protected static function get_style_nodes( $theme_json, $selectors = array() ) {
+		$nodes = array();
+		if ( ! isset( $theme_json['styles'] ) ) {
+			return $nodes;
+		}
+
+		// Top-level.
+		$nodes[] = array(
+			'path'     => array( 'styles' ),
+			'selector' => static::ROOT_BLOCK_SELECTOR,
+		);
+
+		if ( isset( $theme_json['styles']['elements'] ) ) {
+			foreach ( $theme_json['styles']['elements'] as $element => $node ) {
+				$nodes[] = array(
+					'path'     => array( 'styles', 'elements', $element ),
+					'selector' => static::ELEMENTS[ $element ],
+				);
+			}
+		}
+
+		// Blocks.
+		if ( ! isset( $theme_json['styles']['blocks'] ) ) {
+			return $nodes;
+		}
+
+		$nodes = array_merge( $nodes, static::get_block_nodes( $theme_json ) );
+
+		return apply_filters( 'gutenberg_get_style_nodes', $nodes );
```

Comment:
This function is copied from the parent class except this line and the one above.

Reply:
Why do we filter out `filter_out_block_nodes` later and not right here? Since we can override the entire function can't we just change it?